### ISSUE TITLE
SSH.NET does not work for AWS hosted SFTP servers using SFTP protocol

### ISSUE THREAD

**Issue by kikaragyozov** [sshnet/SSH.NET#1199]
Similar issue related to this: https://github.com/sshnet/SSH.NET/issues/1002

It seems the host key they use by default is one that this library does not support. Who's at fault here? I'd really be surprised if AWS are using by default something that is antique or deprecated.

Relevant log entry:

```
{
  "activity-type": "KEX_FAILURE",
  "source-ip": "XXX.XXX.XXX.XXX",
  "resource-arn": "XXXXXXX",
  "message": "no matching host key type found",
  "kex": "ssh-ed25519,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-rsa,ssh-dss"
}
```

The private key being used is an ED25519 one.

**Comment by Rob-Hague** [comment]
Please try with the latest code on the `develop` branch and report back as to whether it works or not.

**Comment by WojciechNagorski** [comment]
Fixed by #1177 and #1180

**Comment by WojciechNagorski** [comment]
Version 2023.0.0 has been published https://www.nuget.org/packages/SSH.NET/2023.0.0
